client/boot/locale.js

```diff
@@ -68,8 +68,8 @@ export const setupLocale = ( currentUser, reduxStore ) => {
 		// For logged out Calypso pages, set the locale from query param
 		const pathLocaleSlug = getLocaleFromQueryParam();
 		pathLocaleSlug && reduxStore.dispatch( setLocale( pathLocaleSlug, '' ) );
-	} else {
-		// For logged out Calypso pages, set the locale from slug
+	} else if ( ! window.hasOwnProperty( 'localeFromPathname' ) ) {
```

Comment:
Do you think we need to add a condition here that sets the locale if `localeFromPathname` is defined? Or would you rather to leave that logic to the page controllers? 🤔 And the `else` can remain as it is.